examples/sine.rs

```diff
@@ -41,15 +41,19 @@ fn main() -> Result<(), coreaudio::Error> {
     // Construct an Output audio unit that delivers audio to the default output device.
     let mut audio_unit = AudioUnit::new(IOType::DefaultOutput)?;
 
-    let stream_format = audio_unit.output_stream_format()?;
+    let stream_format = audio_unit.input_stream_format()?;
```

Comment:
I think this should be `output_stream_format`, no?

On a related note, I wonder if we should rename `input_stream_format` to `capture_stream_format` and `output_stream_format` to `render_stream_callback`? Can probably wait for a future PR though if we decide to do it.

Reply:
This is really confusing in CoreAudio! For me it helps to think of a playback or capture AudioUnit as an extra block that sits between the application and the physical device. Then it becomes logical (I mean slightly less confusing) that the input format of a playback AudioUnit is the format used when the application passes data to it. 
I added a comment in the code here.

Reply:
Ahh thanks for the explanation again, I think this is the second time I've been caught out by this now!